### PR TITLE
✨ Study model

### PR DESCRIPTION
**Pull request by znatty22**
Add study model and one to many relation to participant
Add unit tests for study model 
Add updates to all entity unit tests to support addition of study
Add entity relationship diagram
Add migrations

### CODE REVIEW

File: dataservice/api/participant/models.py [hunk 1]
```diff
@@ -36,3 +36,9 @@ class Participant(db.Model, Base):
                                  cascade="all, delete-orphan",
                                  backref=db.backref('participants',
                                                     lazy=True))
+    study_id = db.Column(db.String(8),
+                         db.ForeignKey('study.kf_id'),
+                         nullable=False)
+
+    def __repr__(self):
+        return '<Participant {}>'.format(self.external_id)
```
Comment (dankolbman) [inline dataservice/api/participant/models.py]: Could we use `kf_id` here?

File: dataservice/api/study/models.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    :param external_id: dbGap accession number
+    :param version: dbGap version
+    :param name: Name or title of study
+    :param attribution: Link to attribution prose provided by dbGap
+    """
+    __tablename__ = "study"
+
+    data_access_authority = db.Column(db.Text(),
+                                      nullable=False,
+                                      default='dbGap')
```
Comment (dankolbman) [inline dataservice/api/study/models.py]: technical stylized as `dbGaP`

File: dataservice/api/study/models.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    version = db.Column(db.Text())
+    name = db.Column(db.Text())
+    attribution = db.Column(db.Text())
+
+    participants = db.relationship(Participant,
+                                   cascade="all, delete-orphan",
+                                   backref='study')
+
+    def __repr__(self):
+        return '<Study {}>'.format(self.external_id)
```
Comment (dankolbman) [inline dataservice/api/study/models.py]: Should also be `kf_id`

File: tests/aliquot/test_aliquot_models.py [hunk 4]
```diff
@@ -20,6 +21,9 @@ def create_participant_sample_aliquot(self):
         create a participant, sample, and aliquot save to db
```
Comment (parimalak) [inline tests/aliquot/test_aliquot_models.py:21]: we could comment something about creating study

File: migrations/versions/c23f9043d57b_.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    sa.Column('data_access_authority', sa.Text(), nullable=False),
+    sa.Column('external_id', sa.Text(), nullable=False),
+    sa.Column('version', sa.Text(), nullable=True),
+    sa.Column('name', sa.Text(), nullable=True),
+    sa.Column('attribution', sa.Text(), nullable=True),
+    sa.PrimaryKeyConstraint('kf_id'),
+    sa.UniqueConstraint('kf_id'),
+    sa.UniqueConstraint('uuid')
+    )
+    op.create_unique_constraint(None, 'outcome', ['kf_id'])
```
Comment (dankolbman) [inline migrations/versions/c23f9043d57b_.py]: Shouldn't be in this migration.

Comment (znatty22) [reply migrations/versions/c23f9043d57b_.py]: Hmm, yea it shouldn't

**Review by dankolbman — APPROVED** [review]
👍

**Review by parimalak — APPROVED** [review]

**Pull request merged**